sound/soc/codecs/rt722-sdca.h

```diff
@@ -0,0 +1,240 @@
+	bool fu0f_mixer_r_mute;
+	bool fu1e_mixer_mute[4];
+};
+
+struct rt722_sdca_dmic_kctrl_priv {
+	unsigned int reg_base;
+	unsigned int count;
+	unsigned int max;
+	unsigned int invert;
+};
+
+struct sdw_stream_data {
+	struct sdw_stream_runtime *sdw_stream;
+};
+
+/* NID */
+#define RT722_VENDOR_REG				0x20
+#define RT722_VENDOR_CALI				0x58
+#define RT722_VENDOR_SPK_EFUSE			0x5c
+#define RT722_VENDOR_IMS_DRE			0x5b
+#define RT722_VENDOR_ANALOG_CTL			0x5f
+#define RT722_VENDOR_HDA_CTL			0x61
+
+/* Index (NID:20h) */
+#define RT722_JD_PRODUCT_NUM			0x00
+#define RT722_ANALOG_BIAS_CTL3			0x04
+#define RT722_JD_CTRL1					0x09
+#define RT722_LDO2_3_CTL1				0x0e
+#define RT722_LDO1_CTL					0x1a
+#define RT722_HP_JD_CTRL				0x24
+#define RT722_CLSD_CTRL6				0x3c
+#define RT722_COMBO_JACK_AUTO_CTL1		0x45
+#define RT722_COMBO_JACK_AUTO_CTL2		0x46
+#define RT722_COMBO_JACK_AUTO_CTL3		0x47
+#define RT722_DIGITAL_MISC_CTRL4		0x4a
+#define RT722_FSM_CTL					0x67
+#define RT722_SDCA_INTR_REC				0x82
+#define RT722_SW_CONFIG1				0x8a
+#define RT722_SW_CONFIG2				0x8b
+
+/* Index (NID:58h) */
+#define RT722_DAC_DC_CALI_CTL0			0x00
+#define RT722_DAC_DC_CALI_CTL1			0x01
+#define RT722_DAC_DC_CALI_CTL2			0x02
+#define RT722_DAC_DC_CALI_CTL3			0x03
+
+/* Index (NID:59h) */
+#define RT722_ULTRA_SOUND_DETECTOR6		0x1e
+
+/* Index (NID:5bh) */
+#define RT722_IMS_DIGITAL_CTL1			0x00
+#define RT722_IMS_DIGITAL_CTL5			0x05
+#define RT722_HP_DETECT_RLDET_CTL1		0x29
+#define RT722_HP_DETECT_RLDET_CTL2		0x2a
+
+/* Index (NID:5fh) */
+#define RT722_MISC_POWER_CTL0				0x00
+#define RT722_MISC_POWER_CTL7				0x08
+
+/* Index (NID:61h) */
+#define RT722_HDA_LEGACY_MUX_CTL0			0x00
+#define RT722_HDA_LEGACY_UNSOL_CTL			0x03
+#define RT722_HDA_LEGACY_CONFIG_CTL0		0x06
+#define RT722_HDA_LEGACY_RESET_CTL			0x08
+#define RT722_HDA_LEGACY_GPIO_WAKE_EN_CTL	0x0e
+#define RT722_DMIC_ENT_FLOAT_CTL			0x10
+#define RT722_DMIC_GAIN_ENT_FLOAT_CTL0		0x11
+#define RT722_DMIC_GAIN_ENT_FLOAT_CTL2		0x13
+#define RT722_ADC_ENT_FLOAT_CTL				0x15
+#define RT722_ADC_VOL_CH_FLOAT_CTL			0x17
+#define RT722_ADC_SAMPLE_RATE_FLOAT			0x18
+#define RT722_DAC03_HP_PDE_FLOAT_CTL		0x22
+#define RT722_MIC2_LINE2_PDE_FLOAT_CTL		0x23
+#define RT722_ET41_LINE2_PDE_FLOAT_CTL		0x24
+#define RT722_ADC0A_08_PDE_FLOAT_CTL		0x25
+#define RT722_ADC10_PDE_FLOAT_CTL			0x26
+#define RT722_DMIC1_2_PDE_FLOAT_CTL			0x28
+#define RT722_AMP_PDE_FLOAT_CTL				0x29
+#define RT722_I2S_IN_OUT_PDE_FLOAT_CTL		0x2f
+#define RT722_GE_RELATED_CTL1				0x45
+#define RT722_GE_RELATED_CTL2				0x46
+#define RT722_MIXER_CTL0					0x52
+#define RT722_MIXER_CTL1					0x53
+#define RT722_EAPD_CTL						0x55
+#define RT722_UMP_HID_CTL0					0x60
+#define RT722_UMP_HID_CTL1					0x61
+#define RT722_UMP_HID_CTL2					0x62
+#define RT722_UMP_HID_CTL3					0x63
+#define RT722_UMP_HID_CTL4					0x64
+#define RT722_UMP_HID_CTL5					0x65
+#define RT722_UMP_HID_CTL6					0x66
+#define RT722_UMP_HID_CTL7					0x67
+#define RT722_UMP_HID_CTL8					0x68
```

Comment:
the tab alignment is difficult to follow, it's not clear if the different indent is intentional or not.

Reply:
will fix it